src/Libraries/CoreNodeModels/Input/ColorPalette.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+
+using Dynamo.Graph;
+using Dynamo.Graph.Nodes;
+
+using ProtoCore.AST.AssociativeAST;
+using CoreNodeModels.Properties;
+
+using DSColor = DSCore.Color;
+
+using Autodesk.DesignScript.Runtime;
+
+
+namespace CoreNodeModels.Input
+{
+    [NodeName("Color Palette")]
+    [NodeDescription("ColorPaletteDescription", typeof(Resources))]
+    [NodeCategory("Core.Color.Create")]
+    [NodeSearchTags("ColorUISearchTags", typeof(Resources))]
+    [IsDesignScriptCompatible]
+    [OutPortNames("Color")]
+    [OutPortTypes("Color")]
+    [OutPortDescriptions("Selected Color.")]
+    public class ColorPalette : NodeModel
+    {
+        private DSColor dscolor = DSColor.ByARGB(255,0,0,0);
+        public DSColor dsColor
+        {
+            get { return dscolor; }
+            set
+            {                              
+                dscolor = value;
+                OnNodeModified();
+            }
+        }
+        /// <summary>
+        ///     Node constructor.
+        /// </summary>
+        public ColorPalette()
+        {
+            RegisterAllPorts();
+        }
+        private DSColor DeserializeValue(string val)
+         {
+            try
+            {
+                var nums = new List<int>();
```

Comment:
refer to the following http://mono.1490590.n4.nabble.com/Mono-Regex-not-working-the-same-as-MS-NET-td1538428.html and https://forums.xamarin.com/discussion/31229/intermittent-failing-unit-tests-for-regular-expressions

I am happy to test a regex if preferred? Then fallback to non regex? My issue is mostly concerned around if it's a weird bug it may be causing different issues on different machines and the reason may not be obvious.
Let me setup my unit tests firsts, if I am getting failures I will ask your advice on how best to proceed.